pkg/services/ssosettings/ssosettingsimpl/service.go

```diff
@@ -114,14 +123,19 @@ func (s *Service) GetForProviderWithRedactedSecrets(ctx context.Context, provide
 }
 
 func (s *Service) List(ctx context.Context) ([]*models.SSOSettings, error) {
-	result := make([]*models.SSOSettings, 0, len(ssosettings.AllOAuthProviders))
+	availableProviders := ssosettings.AllOAuthProviders
+	if s.licensing.FeatureEnabled(social.SAMLProviderName) {
+		availableProviders = ssosettings.AllProviders
+	}
```

Comment:
Same as above, I don't think we should store the license in the service struct, instead keep a list of available providers. You might need 2 lists, one containing the configurable providers, the other containing all of them.